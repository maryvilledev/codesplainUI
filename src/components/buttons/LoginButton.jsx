import React from 'react';
import { FlatButton } from 'material-ui';
import cookie from 'react-cookie'
import GitHubIcon from '../../../res/github-icon.png';

const CLIENT_ID = process.env.REACT_APP_CLIENT_ID
const GITHUB_URL = `https://github.com/login/oauth/authorize?client_id=${CLIENT_ID}`

const styles = {
  button: {
    marginTop: '6px',
  },
  span: {
    color: 'white',
  },
  img: {
    marginRight: '7px',
  },
}

// Saves a cookie containing the current URL path, so we can retrieve it
// later when GitHub sends the user back to the app, and send the user back
// to the page they were on when they clicked this button.
const saveSignInRedirect = () => {
  cookie.save('signInRedirect', location.pathname, { path: '/' });
}

/*
<LoginButton /> renders as white clickable "LOG IN" text, with a white octocat
icon to its left. When clicked, the button sends the user to the GitHub OAuth 
authorization URL, where the client ID is set to the CLIENT_ID env var.

If the 'href' prop is specified, then the button will redirect to that URL 
instead of the GitHub OAuth one.
*/
const LoginButton = ({ href }) => (
  <FlatButton 
    href={href ? href : GITHUB_URL}
    onClick={saveSignInRedirect}
    style={styles.button}
    label={
      <span style={styles.span}>
        <img 
          role="presentation"
          width="23" 
          style={styles.img} 
          src={GitHubIcon} 
        />
        Log In
      </span>
    }
  />
);

export default LoginButton
