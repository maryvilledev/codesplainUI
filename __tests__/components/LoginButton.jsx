import React from 'react';
import ReactTestUtils from 'react-addons-test-utils';

import LoginButton from '../../src/components/buttons/LoginButton';

describe('<LoginButton />', () => {
  it('matches the stored snapshot', () => {
    const renderer = ReactTestUtils.createRenderer();
    const tree = renderer.render(
      <LoginButton href='dummy-link' />
    );
    expect(tree).toMatchSnapshot();
  });
});
