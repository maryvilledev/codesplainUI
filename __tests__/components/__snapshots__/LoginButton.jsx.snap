exports[`<LoginButton /> matches the stored snapshot 1`] = `
<FlatButton
  disabled={false}
  href="dummy-link"
  label={
    <span
      style={
        Object {
          "color": "white",
        }
      }>
      <img
        role="presentation"
        src="github-icon.png"
        style={
          Object {
            "marginRight": "7px",
          }
        }
        width="23" />
      Log In
    </span>
  }
  labelPosition="after"
  labelStyle={Object {}}
  onClick={[Function]}
  onKeyboardFocus={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onTouchStart={[Function]}
  primary={false}
  secondary={false}
  style={
    Object {
      "marginTop": "6px",
    }
  } />
`;
